pkg/provenance/generate.go

```diff
@@ -144,51 +159,57 @@ func generateSignedAttestation(it in_toto.Statement, appDirPath string) {
 	x509Encoded, err := encrypted.Decrypt(pb.Bytes, []byte(pwd))
 
 	if err != nil {
-		log.Info("Error in dycrypting private key")
-		return
+		log.Errorf("Error in dycrypting private key: %s", err.Error())
+		return err
 	}
 	priv, err := x509.ParsePKCS8PrivateKey(x509Encoded)
 
 	if err != nil {
-		log.Info("Error in parsing private key")
-		return
+		log.Errorf("Error in parsing private key: %s", err.Error())
+		return err
 	}
 
 	signer, err := ssl.NewEnvelopeSigner(&IntotoSigner{
 		priv: priv.(*ecdsa.PrivateKey),
 	})
 	if err != nil {
-		log.Info("Error in creating new signer")
-		return
+		log.Errorf("Error in creating new signer: %s", err.Error())
+		return err
 	}
 
 	env, err := signer.SignPayload("application/vnd.in-toto+json", b)
 	if err != nil {
-		log.Info("Error in signing payload")
-		return
+		log.Errorf("Error in signing payload: %s", err.Error())
+		return err
 	}
 
 	// Now verify
 	err = signer.Verify(env)
 	if err != nil {
-		log.Info("Error in verifying env")
-		return
+		log.Errorf("Error in verifying env: %s", err.Error())
+		return err
 	}
 
 	eb, err := json.Marshal(env)
 	if err != nil {
-		log.Info("Error in marshaling env")
-		return
+		log.Errorf("Error in marshaling env: %s", err.Error())
+		return err
 	}
 
 	log.Debug("attestation.json", string(eb))
 
-	utils.WriteToFile(string(eb), appDirPath, utils.ATTESTATION_FILE_NAME)
+	err = utils.WriteToFile(string(eb), appDirPath, utils.ATTESTATION_FILE_NAME)
+	if err != nil {
+		log.Errorf("Error in writing attestation to a file: %s", err.Error())
+		return err
+	}
 
 	attestationPath := filepath.Join(appDirPath, utils.ATTESTATION_FILE_NAME)
 
 	upload(it, attestationPath)
```

Comment:
thank you for adding error handlings to many other parts,
but we also need an error handling for this function because this can fail for many possible reasons.

Reply:
thank you for pointing that out. I think I still need to handle it.